utils/include/detray/simulation/simulator.hpp

```diff
@@ -100,7 +99,7 @@ struct simulator {
     unsigned int m_events = 0;
     std::string m_directory = "";
     std::unique_ptr<detector_t> m_detector;
-    track_generator_t m_track_generator;
+    track_generator_t& m_track_generator;
```

Comment:
Hm, the simulator seems to be a data owning class, so I don't think a reference member would be a good idea. Is it possible to use a smart pointer and move the track generator into the simulator, so that the simulator will not depend on the track generator lifetime? We might need to add a move constructor

Reply:
Seems possible. Could you check?